File: no-auth/src/main/java/com/kjj/noauth/dto/user/UserInfoDto.java
package com.kjj.noauth.dto.user;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class UserInfoDto {
    private Long id;
    private String username;

    public static UserInfoDto from(UserDto userDto) {
        return new UserInfoDto(
                userDto.getId(),
                userDto.getUsername()
        );
    }

    public static UserInfoDto loginFail(String message) {
        return new UserInfoDto(
                null,
                message
        );
    }
}

File: no-auth/src/main/kotlin/com/kjj/noauth/dto/user/UserInfoDto.kt
package com.kjj.noauth.dto.user

data class UserInfoDto(
    val id: Long,
    val username: String
) {
    companion object {
        fun from(userDto: UserDto): UserInfoDto {
            return UserInfoDto(
                userDto.id!!,
                userDto.username
            )
        }

        fun loginFail(message: String): UserInfoDto {
            return UserInfoDto(
                0,
                message
            )
        }
    }
}